docs/src/design/stack/crypto_ops.md

```diff
@@ -0,0 +1,122 @@
+# Cryptographic operations
+In this section we describe the AIR constraint for Miden VM cryptographic operations.
+
+Cryptographic operations in Miden VM are performed by the [Hash chiplet](../chiplets/hasher.md). Communication between the stack and the hash chiplet is accomplished via the chiplet bus $b_{chip}$. To make requests to and to read results from the chiplet bus we need to divide its current value by the value representing the request.
+
+Thus, to describe AIR constraints for the cryptographic operations, we need to define how to compute these input and output values within the stack. We do this in the following sections.
+
+## RPPERM
+The `RPPERM` operation applies Rescue Prime permutation to the top $12$ elements of the stack. The stack is assumed to be arranged so that the $8$ elements of the rate are at the top of the stack. The capacity word follows, with the number of elements to be hashed at the deepest position in stack. The diagram below illustrates this graphically.
+
+![rpperm](../../assets/design/stack/crypto_ops/RPPERM.png)
+
+In the above, $r$ (located in the helper register $h_0$) is the row address from the hash chiplet set by the prover nondeterministically.
+
+For the `RPPERM` operation, we define input and output values as follows:
+
+$$
+v_{input} = \alpha_0 + \alpha_1 \cdot op_{linhash} + \alpha_2 \cdot h_0 + \sum_{j=0}^{11} (\alpha_{j+4} \cdot s_{11-i})
+$$
+
+$$
+v_{output} = \alpha_0 + \alpha_1 \cdot op_{retstate} + \alpha_2 \cdot (h_0 + 7) + \sum_{j=0}^{11} (\alpha_{i+4} \cdot s_{11-i}')
+$$
+
+In the above, $op_{linhash}$ and $op_{retstate}$ are the unique [operation label](../chiplets/main.md#operation-labels) for initiating a linear hash and reading the full state of the hasher respectively. Also note that the term for $\alpha_3$ is missing from the above expressions because for Rescue Prime permutation computation the index column is expected to be set to $0$.
```

Comment:
nit: "... are the unique operation labels..." (this comes up a few times, so if you do want to change it then maybe just search & replace. I won't mark the others)

Reply:
Fixed!